### PR TITLE
functional vpc resource

### PR DESCRIPTION
**Pull request by samcornwell**
Filled in original vpc skeleton to make it functional.
  
#145

### CODE REVIEW

File: docs/resources/aws_vpc.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+### instance_tenancy
+
+The allowed tenancy of instances launched into the VPC.
+
+    describe aws_vpc do
+      its ('instance_tenancy') { should eq 'default' }
+    end
+
+###  is_default
```
Comment (clintoncwolfe) [inline docs/resources/aws_vpc.md]: We generally don't document "matchers-as-properties" - we want the user to use the matcher.

File: libraries/aws_vpc.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    if @vpc_id.nil?
+      filter = { name: 'isDefault', values: ['true'] }
+    else
+      filter = { name: 'vpc-id', values: [@vpc_id] }
+    end
+
+    resp = backend.describe_vpcs({ filters: [filter] })
+
+    @vpc = resp.vpcs[0].to_h
```
Comment (clintoncwolfe) [inline libraries/aws_vpc.rb]: Will this fail if the search is a miss?

File: test/integration/default/verify/controls/aws_vpc.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  'ec2_security_group_default_vpc_id',
+].each do |fixture_name|
+  fixtures[fixture_name] = attribute(
+    fixture_name,
+    default: "default.#{fixture_name}",
+    description: 'See ../build/ec2.tf',
+  )
+end
+
+control "aws_vpc default vpc" do
```
Comment (clintoncwolfe) [inline test/integration/default/verify/controls/aws_vpc.rb]: Please split into sections: searching, then exercising properties and matchers.

File: test/unit/resources/aws_vpc_test.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+  def test_rejects_unrecognized_params
+    assert_raises(ArgumentError) { AwsVpc.new(shoe_size: 9) }
+  end
+
+  def test_rejects_invalid_vpc_id
+    assert_raises(ArgumentError) { AwsVpc.new('vpc-rofl') }
+  end
+end
+
```
Comment (clintoncwolfe) [inline test/unit/resources/aws_vpc_test.rb]: Need unit tests for search, properties, and matchers.

File: test/integration/default/verify/controls/aws_vpc.rb (hunk 3 above)
Comment (clintoncwolfe) [reply test/integration/default/verify/controls/aws_vpc.rb]: I think we need to make a non-default VPC so we can test the is_default matcher.

Comment (clintoncwolfe) [reply test/integration/default/verify/controls/aws_vpc.rb]: We need a search miss test.

File: test/integration/default/verify/controls/aws_vpc.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  describe aws_vpc(fixtures['ec2_security_group_default_vpc_id']) do
```
Comment (clintoncwolfe) [inline test/integration/default/verify/controls/aws_vpc.rb]: This searches for the default VPC, using the vpc_id that was provided via the fixtures mechanism.  We should also test getting the default VPC by not providing any params.

**Review by clintoncwolfe — CHANGES_REQUESTED** [review]
I liked the approach you took with the `@vpc` instance variable.  

Mainly missing some testing.

File: libraries/aws_vpc.rb (hunk 2 above)
Comment (samcornwell) [reply libraries/aws_vpc.rb]: It won't throw an exception.  `@vpc` will be an empty hash, `@vpc_id` will be `nil`, and `@exists` will be `false`.

**Comment by samcornwell** [comment]
@clintoncwolfe Ok, I believe I have addressed all of your concerns. I replied inline to your comments on libraries/aws_vpc.rb.

**Comment by clintoncwolfe** [comment]
Excellent work!  Next thing is for me to run the tests.

**Comment by samcornwell** [comment]
@clintoncwolfe Btw I just committed a tiny change to rename two of the unit tests.

**Review by clintoncwolfe — APPROVED** [review]

**Comment by clintoncwolfe** [comment]
Very nice PR - thank you for your contribution!  Tests run clean.

**Comment by aaronlippold** [comment]
@samcornwell good job!

**Pull request merged**